### PR TITLE
implement metaclass support in the interpreter

### PR DESCRIPTION
**Pull request by t-vi**
fixes: #831

### CODE REVIEW

File: thunder/core/interpreter.py [hunk 1]
```diff
@@ -2671,6 +2671,25 @@ def create_namedtuple(typename: str, field_names: str, **kwargs):
...
+    if obj is INTERPRETER_SIGNALS.EXCEPTION_RAISED:
+        return obj
+
+    wrapped_init = _interpret_call(getattr, obj, wrap_const("__init__"))
+    assert not isinstance(wrapped_init, INTERPRETER_SIGNALS)
+    populate_attribute_wrapper(wrapped_init, "__self__", obj)
+    res = _interpret_call(wrapped_init, *args, **kwargs)
+    if res is INTERPRETER_SIGNALS.EXCEPTION_RAISED:
+        return res
+    return obj
```
Comment (nikitaved) [inline thunder/core/interpreter.py:2690]: I am a bit confused with these lines. After successful init we still return the result of new?

Comment (t-vi) [reply thunder/core/interpreter.py:2690]: Yes, init returns None.

Comment (nikitaved) [reply thunder/core/interpreter.py:2690]: Yeah, makes sense :)

**Review by nikitaved — APPROVED** [review]
LGTM! :)

**Pull request merged**